pkg/apis/monitoring/v1/servicemonitor_types.go

```diff
@@ -29,7 +29,19 @@ const (
 // +k8s:openapi-gen=true
 // +kubebuilder:resource:categories="prometheus-operator",shortName="smon"
 
-// ServiceMonitor defines monitoring for a set of services.
+// The `ServiceMonitor` custom resource definition (CRD) allows to declaratively define how a dynamic set of services should be monitored. Which services are selected to be monitored with the desired configuration is defined using label selections. This allows an organization to introduce conventions around how metrics are exposed, and then following these conventions new services are automatically discovered, without the need to reconfigure the system.
+//
+// For Prometheus to monitor any application within Kubernetes an `Endpoints` object needs to exist. `Endpoints` objects are essentially lists of IP addresses. Typically an `Endpoints` object is populated by a `Service` object. A `Service` object discovers `Pod`s by a label selector and adds those to the `Endpoints` object.
```

Comment:
Do we want to detail this much detail on api doc? I think this will better suit for documentation?

Reply:
Maybe line 32 might be enough. Wdyt?

Reply:
PodMonitor also has lot of detail. Maybe we can remove some lines there too?

Reply:
I think ``ServiceMonitor` allows to declaratively define how a dynamic set of services should be monitored. Which services are selected to be monitored with the desired configuration is defined using label selections. `` should be enough